go/cache/cache.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+// DefaultCacheSize is the default size for a Vitess cache instance.
+// If this value is specified in BYTES, Vitess will use a LFU-based cache that keeps track of the total
+// memory usage of all cached entries accurately. If this value is specified in ENTRIES, Vitess will
+// use the legacy LRU cache implementation which only tracks the amount of entries being stored.
+// Changing this value affects:
+// - the default values for CLI arguments in VTGate
+// - the default values for the config files in VTTablet
+// - the default values for the test caches used in integration and end-to-end tests
+// Regardless of the default value used here, the user can always override Vitess' configuration to
+// force a specific cache type (e.g. when passing a value in ENTRIES to vtgate, the service will use
+// a LRU cache).
+const DefaultCacheSize = SizeInEntries(5000)
```

Comment:
This is the global toggle that will allow us to switch the default cache implementation.

Reply:
This is clever! Correct me if I'm wrong, but it seems to me that if we toggle this, then the [code](https://github.com/vitessio/vitess/pull/7439/files#diff-386dafefbaf1d87f35fb8c081ef12aed4af6aeb89db721806b21dacc8c49a438R183-R187) in `vtgate.go` would have to change to allow use of the `Entries` option.

Reply:
Correct! I should probably flag that. The code in `vtgate.go` is written so that when the LRU cache is the default, setting a LFU cache flag overrides it. The opposite needs to be true when the LFU cache is the default.

Reply:
@deepthi Fixed in https://github.com/vitessio/vitess/pull/7439/commits/a3feeb8eca2311d0cd04e756c18971ddb33475c8 so we don't have to change this in the future :ok_hand: